frontend/src/timeline.tsx

```diff
@@ -0,0 +1,151 @@
+import classes from './timeline.module.css';
+import Event from './Event/Event';
+import {
+  DayInfo, 
+  TimelineInfo
+} from '../../backend/src/interfaces';
+
+const dayMonthOption = { day: 'numeric', month: 'numeric' };
+const weekdayOption = { weekday: 'long' };
+
+const getDate = function (date: Date, options: { }): string {
+  return date.toLocaleDateString('en-GB', options);
+};
+
+const startDate = new Date(2024, 9, 28);
+const endDate = new Date(2024, 10, 3);
+const days = Array.from({ length: 7 }, (e, i) => {
+
+  e = 50; //straight up this line is just here to get rid of "value declared but not read" warning. 
+  // it was bugging me
+
+  const date: Date = new Date(startDate.valueOf());
+
+  date.setDate(startDate.getDate() + i);
+
+  return {
+    date,
+    events: []
+  };
+});
+
+function Day(dayProps: DayInfo) {
```

Comment:
this component, as well as the others in this file, should probably be in their own aptly-named files so it's easy to find what we're looking for